Bottom: d56cf047a62e1d834f93eec581ff861efff6b169
Top:    fe4cb17f4c0c26ecadfff46a5a44815176c4acbd
Author: Ian Kent <raven@themaw.net>
Date:   2013-03-01 15:37:45 +0800

autofs-5.0.7 - fix map read fail incorrectly set on master re-read

When re-reading the master map read failures for plus included maps
should not cause the map read to fail, they should be ingored.


---

diff --git a/CHANGELOG b/CHANGELOG
index 39388a5..b13a5f5 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -28,6 +28,7 @@
 - make yellow pages support optional.
 - modules/replicated.c: use sin6_addr.s6_addr32.
 - workaround missing GNU versionsort extension.
+- fix map read fail incorrectly set on master re-read.
 
 25/07/2012 autofs-5.0.7
 =======================
diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index facb305..b6432ff 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -443,7 +443,7 @@ int lookup_read_master(struct master *master, time_t age, void *context)
 
 			inc = check_master_self_include(master, ctxt);
 			if (inc) 
-				master->recurse = 1;;
+				master->recurse = 1;
 			master->depth++;
 			status = lookup_nss_read_master(master, age);
 			if (!status) {
@@ -452,6 +452,11 @@ int lookup_read_master(struct master *master, time_t age, void *context)
 				     "failed to read included master map %s",
 				     master->name);
 			}
+			/*
+			 * Plus map include failures don't cause the map
+			 * read to fail.
+			 */
+			master->read_fail = 0;
 			master->depth--;
 			master->recurse = 0;
